src/Microsoft.Extensions.Configuration.UserSecrets/ConfigurationExtensions.cs

```diff
@@ -14,8 +13,13 @@ public static class ConfigurationExtensions
         /// </summary>
         /// <param name="configuration"></param>
         /// <returns></returns>
-        public static IConfigurationBuilder AddUserSecrets([NotNull]this IConfigurationBuilder configuration)
+        public static IConfigurationBuilder AddUserSecrets(this IConfigurationBuilder configuration)
         {
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
```

Comment:
Agreed, but name `configuration` is not visible to the user of the extension method. To the user, it mostly looks like a normal function (except the `(extension)` text in front of the signature): 

![image](https://cloud.githubusercontent.com/assets/1808428/10505162/1d6fb95a-72bf-11e5-9991-c9d9af482304.png)